lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1010,16 +1010,17 @@ ClangImporter::create(ASTContext &ctx,
 
   // Set up the file manager.
   {
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS;
     if (!ctx.SearchPathOpts.VFSOverlayFiles.empty()) {
       // If the clang instance has overlays it means the user has provided
       // -ivfsoverlay options and swift -vfsoverlay options.  We're going to
       // clobber their file system with our own, so warn about it.
       if (!instance.getHeaderSearchOpts().VFSOverlayFiles.empty()) {
         ctx.Diags.diagnose(SourceLoc(), diag::clang_vfs_overlay_is_ignored);
       }
-      instance.setVirtualFileSystem(ctx.SourceMgr.getFileSystem());
+      VFS = ctx.SourceMgr.getFileSystem();
```

Comment:
I think you can/should do this unconditionally, even if there aren't any Swift-side VFS overlays.

Reply:
I tried that. It might work but I'm seeing 22 failing tests and will need to triage them to make sure they're not related to this change.

Reply:
Hm, that's concerning. That implies that we're stomping on Clang setup when we're using a VFS, and that's not a commonly tested case to begin with.

Reply:
Some of the test failures happen even with the change here, and I don't think they have anything to do with this change. With the change that Jonas proposed in https://github.com/apple/swift/pull/23659, I see test regressions in:

Frontend/vfs.swift
IDE/print_clang_header.swift
SourceKit/CompileNotifications/diagnostics.swift
Frontend/clang-args-diags.swift

I'm going ahead with the change here, at least to get the build working again.